python/tests/sbp/generator.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (C) 2021 Swift Navigation Inc.
+# Contact: https://support.swiftnav.com
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+"""
+Generate missing test cases automatically.
```

Comment:
I am aware that there is `generator/sbpg/generator.py` . This could be merged there , but that would involve sbpg depending on `sbp` which seems undesirable (possibly an optional dep?), and moving some of the test runner logic into sbpg or sbp.

I am wondering if auto generation of test cases is already possible with sbpg or some other tool; it didnt seem so.

There is still a bug in the currently logic.  The generated test for `MsgNetworkBandwidthUsage` fails `assert_package`.

After that, the goal is to not only look for missing tests, but also to measure the coverage of `assert_package` for each existing test package, and look for incoming msgs which provide new coverage.  This will provide test cases for the inner msg classes that are currently not being exercised because the existing tests are often for simpler instances of the outer "real" message types.

Reply:
This looks great, if it becomes a lot of work to resolve the final bug let's split this off into a separate PR.